source/ref_connections.rst

```diff
@@ -0,0 +1,339 @@
+    * You have created the IAM keys for the service account. `Creating and managing service account keys <https://cloud.google.com/iam/docs/creating-managing-service-account-keys>`_.
+
+.. note::
+
+    Workload Identity is the recommended way to access Google Cloud services from within GKE due to its improved security properties and manageability.
+    We will support the Workload Identity in the near future.
+
+The following fields of connection API are required:
+
+    * ``spec.type`` - It must be equal **gcs**.
+    * ``spec.keySecret`` - a service account key in json format.
+    * ``spec.uri`` -  GCS compatible URI, for example gcs://<bucket-name>/dir1/dir2/
+    * ``spec.region`` - `GCP Region <https://cloud.google.com/compute/docs/regions-zones>`_, where a bucket was created.
+
+.. code-block:: yaml
+    :caption: Example of Connection GCS:
+    :name: Connection GCS file
+
+    id: "training-data"
+    spec:
+        type: gcs
+        uri: gsc://raw-data/model/input
+        keySecret: '{"type": "service_account", "project_id": "project_id", "private_key_id": "private_key_id", "private_key": "-----BEGIN PRIVATE KEY-----\nprivate_key\n-----END PRIVATE KEY-----\n", "client_email": "test@project_id.iam.gserviceaccount.com", "client_id": "123455678", "auth_uri": "https://accounts.google.com/o/oauth2/auth", "token_uri": "https://oauth2.googleapis.com/token", "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs", "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/test@project_id.iam.gserviceaccount.com"}'
+        description: "Training data for a model"
+        region: us-central2
+
+Azure Blob storage
+------------------
+
+Odahu-flow uses the `Blob storage <https://docs.microsoft.com/ru-ru/azure/storage/blobs/storage-blobs-introduction>`_ in Azure to store:
+
+    * model trained artifacts.
+    * input data for ML models.
+
+Before usage, make sure that:
+
+    * You have created a storage account . `Create a storage account <https://docs.microsoft.com/ru-ru/azure/storage/blobs/storage-quickstart-blobs-cli#create-a-storage-account>`_.
+    * You have created a storage container in the storage account . `Create a container <https://docs.microsoft.com/ru-ru/azure/storage/blobs/storage-quickstart-blobs-cli#create-a-container>`_.
+    * You have created an SAS token. `Create an account SAS <https://docs.microsoft.com/ru-ru/rest/api/storageservices/create-account-sas>`_.
+
+The following fields of connection API are required:
+
+    * ``spec.type`` - It must be equal **azureblob**.
+    * ``spec.keySecret`` - Odahu-flow uses the `shared access signatures <https://docs.microsoft.com/en-us/azure/storage/common/storage-sas-overview>`_ to authorize in Azure.
+      The key has the following format: "<primary_blob_endpoint>/<sas_token>".
+    * ``spec.uri`` -  GCS compatible URI, for example <bucket-name>/dir1/dir2/
```

Comment:
* ``spec.uri`` -  Azure storage compatible URI, for example <bucket-name>/dir1/dir2/